pkg/ovs/openflow/ofctrl_bridge.go

```diff
@@ -304,32 +304,78 @@ func (b *OFBridge) IsConnected() bool {
 	return b.ofSwitch.IsReady()
 }
 
-func (b *OFBridge) AddFlowsInBundle(addflows []Flow, modFlows []Flow, delFlows []Flow) error {
+const (
+	messageAdd int = iota
+	messageModify
+	messageDelete
+)
+
+func (b *OFBridge) AddOFEntriesInBundle(addEntries []OFEntry, modEntries []OFEntry, delEntries []OFEntry) error {
 	// If no Openflow entries are requested to be added or modified or deleted on the OVS bridge, return immediately.
-	if len(addflows) == 0 && len(modFlows) == 0 && len(delFlows) == 0 {
+	if len(addEntries) == 0 && len(modEntries) == 0 && len(delEntries) == 0 {
 		klog.V(2).Info("No Openflow entries need to be synced to the OVS bridge, returning")
 		return nil
 	}
-	// Create a new transaction.
-	tx := b.ofSwitch.NewTransaction(ofctrl.Atomic)
+	type entryOperation struct {
+		entry         OFEntry
+		operationType int
+	}
+	var addFlows, modFlows, delFlows, addGroups, modGroups, delGroups []entryOperation
+	// Classifier the entries according to the EntryType, and set a correct operation type.
+	checkMessages := func(entries []OFEntry, operation int) {
```

Comment:
Is it common to mix flows and groups in []ofEntries? Wonder if we can avoid this re-classification (if that will make the caller code complex, the current way is fine to me too).

Reply:
Ok. We can think about optimizations later.

Reply:
Thought again - do you think we can have two funcs, one needs not re-classification, and another does? Maybe it is not worthwhile, but the extra cost to iterate and regroup flows bother me a little..

Reply:
@jianjuns I guess your point is, we keep the original function `AddFlowsInBundle`, and add a new function `AddEntriesInBundle`. Then we could keep the original callers which are only focusing on Flow entries the same, and the implementation for Service calls the new function, right?

@weiqiangt What do you think? If we add a specific function to support adding Group and Flow in one bundle, is there any impac ont the implementation for Service?

Reply:
Or maybe add a method to OFEntry to convert it to entryOperation, then you don't need this classification and can simply convert OFEntries in a loop. It should be not hard to ask clients put group entry ahead of flow entry.